common/src/main/java/org/apache/uniffle/common/netty/TransportFrameDecoder.java

```diff
@@ -67,9 +67,17 @@ public void channelRead(ChannelHandlerContext ctx, Object data) {
       if (frame == null) {
         break;
       }
-      Message msg = Message.decode(curType, frame);
-      if (msg.body() == null) {
-        frame.release();
+      Message msg = null;
+      try {
+        msg = Message.decode(curType, frame);
+      } finally {
+        boolean shouldRelease =
+            msg != null
```

Comment:
Could you extract a method for this logic? It's easy to test, too.

Reply:
You mean like this?
```
private static boolean isShouldRelease(Message msg) {
    boolean shouldRelease =
        msg == null
            || (msg != null
                && (msg.body() == null
                    || (msg.body().byteBuf() != null
                        && msg.body().byteBuf().readableBytes() == 0)));
    return shouldRelease;
  }
```

Reply:
For testing purposes, package-available is enough. We don't need `public.`

Reply:
> You mean like this?
> 
> ```
> private static boolean shouldRelease(Message msg) {
>     boolean shouldRelease =
>         msg == null
>             || (msg != null
>                 && (msg.body() == null
>                     || (msg.body().byteBuf() != null
>                         && msg.body().byteBuf().readableBytes() == 0)));
>     return shouldRelease;
>   }
> ```
> 
> For testing purposes, maybe private -> public?

You can simplify the code like 
private static boolean shouldRelease(Message msg) {
   if (msg == null) {
        return true;
    }
    if (mgs.body() == null) {
        return true;
   }
    boolean shouldRelease = (msg.body().byteBuf() != null
                        && msg.body().byteBuf().readableBytes() == 0)));
    return shouldRelease;
  }

Reply:
Done. Also, I remove the `private` keyword to make it package available.
@jerqi

Reply:
Could you add a UT for this pr?

Reply:
UTs added. @jerqi 
Also, I have conveniently fixed an issue where a method parameter was missing. The `error ` parameter was forgotten to be passed to the `UnsupportedOperationException`.
https://github.com/apache/incubator-uniffle/blob/master/common/src/main/java/org/apache/uniffle/common/netty/protocol/ResponseMessage.java#L32